### PR TITLE
Add rock paper scissors module

### PR DESCRIPTION
**Pull request by DEVTomatoCake**
Adds the suggestion from https://features.sc-network.net/board/custombot/240
Translations are prepared, they just have to be added in Weblate.

This PR also updates the node-schedule package ([no real changes, only security fix](https://github.com/node-schedule/node-schedule/blob/master/CHANGELOG.md#211---11-january-2023)) and eslint, because why not.
It also fixes a few small typos in the README and increases the time for paginated buttons menus from 20s to 40s. Still too low but may prevent users spamming the buttons to get to a page in time.
If any of those changes above are somehow unwanted, feel free to revert them.

### CODE REVIEW

**Comment by SCDerox** [comment]
Thanks so much for your contribution!
Please add localization directly in the `locales` directory - it will get synced to our weblate once we roll-out your changes on SCNX. We do not add strings directly to weblate.

Once ready, please request a review from me.

*Please note that rollout on SCNX depends on our sole discretion and it may take up to two weeks to reach our main branch - beta branches usually receive your changes after merge).*

Because this is your first PR: If you want to receive a small bounty (= small amount of money) after we've merged your PR as part of our [Developer Programm](https://faq.scnx.app/open-source-developer-pool/), please [create an SCNX Marketplace Organization](https://scnx.app/de/user/organizations/), enable payouts and make sure your Discord-Account is linked to your GitHub.

**Comment by DEVTomatoCake** [comment]
Well, as I can't request a review, take this comment as request for it.

Also not sure how to get the numeral id of the org, doesn't look like it's the name.

**Comment by SCDerox** [comment]
You should be able to request a review in GitHub 🤔

Anyway, to get the ID of your org, please visit https://scnx.app/api/marketplace/organizations/<yourslughere>

**Comment by DEVTomatoCake** [comment]
> You should be able to request a review in GitHub 🤔
Where .-.

File: locales/de.json [hunk 1]
```diff
@@ -865,5 +865,21 @@
...
+        "won": "gewonnen",
+        "lost": "verloren",
+        "tie": "Unentschieden",
+        "play-again": "Erneut spielen",
+        "challenge-message": "%t, %u hat dich zu einer Runde Schere Stein Papier herausgefordert! Klicke auf den Knopf unter dieser Nachricht, um beizutreten! Diese Einladung wird in etwa 2 Minuten ablaufen, also zögere nicht, sie anzunehmen.",
+        "invite-expired": "Entschuldigung, %u, %i hat deine Einladung, Schere Stein Papier zu spielen, nicht rechtzeitig angenommen ):",
+        "invite-denied": "Entschuldigung, %u, aber %i hat deine Einladung, Schere Stein Papier zu spielen, abgelehnt ):",
+        "rps-title": "Schere Stein Papier",
+        "rps-description": "Wähle deine Waffe!",
+        "its-a-tie-try-again": "Untentschieden! Versuch's nochmal!",
```
Comment (SCDerox) [inline locales/de.json:882]:
```suggestion
        "its-a-tie-try-again": "Unentschieden! Versuch's nochmal!",
```
typo

File: modules/rock-paper-scissors/module.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "en": "Let your users play Rock Paper Scissors against the bot and each other!",
+    "de": "Lasse Nutzer auf deinem Server Schere Stein Paper gegen den Bot und gegeneinander spielen"
+  },
+  "commands-dir": "/commands",
+  "noConfig": true,
+  "releaseDate": "0",
+  "tags": [
+    "fun"
+  ],
+  "openSourceURL": "https://github.com/SCNetwork/CustomDCBot/tree/main/modules/rock-paper-scissors"
```
Comment (SCDerox) [inline modules/rock-paper-scissors/module.json:22]:
```suggestion
  "openSourceURL": "https://github.com/DEVTomatoCake/ScootKit-CustomBot/tree/main/modules/rock-paper-scissors"
```
We suggest using your own URL here

File: modules/rock-paper-scissors/commands/rock-paper-scissors.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+                        win2 = localize('rock-paper-scissors', 'lost');
+                    } else if (resU2 === moves[j] && resU1 === movesDouble[j + 2]) {
+                        win1 = localize('rock-paper-scissors', 'lost');
+                        win2 = localize('rock-paper-scissors', 'won');
+                    }
+                }
+            }
+
+            game.state1 = win1;
+            game.state2 = win2;
```
Comment (SCDerox) [inline modules/rock-paper-scissors/commands/rock-paper-scissors.js:254]: `win1` and `win2` are only declared implicitly.
```suggestion
        let win1, win2;
        if (user2.bot) {
            const picked = moves[Math.floor(Math.random() * moves.length)];

            if (i.customId === 'rps_stone') resU1 = moves[0];
            else if (i.customId === 'rps_paper') resU1 = moves[1];
            else if (i.customId === 'rps_scissors') resU1 = moves[2];

            if (picked === resU1) {
                win1 = localize('rock-paper-scissors', 'tie');
                win2 = localize('rock-paper-scissors', 'tie');
            } else {
                for (let j = 0; j < moves.length; j++) {
                    if (picked === moves[j] && resU1 === movesDouble[j + 1]) {
                        win1 = localize('rock-paper-scissors', 'won');
                        win2 = localize('rock-paper-scissors', 'lost');
                    } else if (picked === moves[j] && resU1 === movesDouble[j + 2]) {
                        win1 = localize('rock-paper-scissors', 'lost');
                        win2 = localize('rock-paper-scissors', 'won');
                    }
                }
            }

            game.state1 = win1;
            game.state2 = win2;
            rpsgames[i.message.id] = game;

            if (picked === resU1) embed.setTitle(localize('rock-paper-scissors', 'its-a-tie-try-again'));
            else embed.setTitle(localize('rock-paper-scissors', 'rps-title'));
            embed.setDescription('<@' + game.user1.id + '>: **' + resU1 + '**' + (resU1 !== picked ? ' (' + win1 + ')' : '') + '\n<@' + game.user2.id + '>: **' + picked + '**' + (resU1 !== picked ? ' (' + win2 + ')' : ''));

            if (picked === resU1) components = resetGame(game);
            else components = [generatePlayer(game.user1, game.user2, win1, win2), playagain()];
        } else {
            let resU2 = '';
            if (game.selected1 === 'rps_stone') resU2 = moves[0];
            else if (game.selected1 === 'rps_paper') resU2 = moves[1];
            else if (game.selected1 === 'rps_scissors') resU2 = moves[2];

            if (game.selected2 === 'rps_stone') resU1 = moves[0];
            else if (game.selected2 === 'rps_paper') resU1 = moves[1];
            else if (game.selected2 === 'rps_scissors') resU1 = moves[2];

            if (resU1 === resU2) {
                win1 = localize('rock-paper-scissors', 'tie');
                win2 = localize('rock-paper-scissors', 'tie');
            } else {
                for (let j = 0; j < moves.length; j++) {
                    if (resU2 === moves[j] && resU1 === movesDouble[j + 1]) {
                        win1 = localize('rock-paper-scissors', 'won');
                        win2 = localize('rock-paper-scissors', 'lost');
                    } else if (resU2 === moves[j] && resU1 === movesDouble[j + 2]) {
                        win1 = localize('rock-paper-scissors', 'lost');
                        win2 = localize('rock-paper-scissors', 'won');
                    }
                }
            }

            game.state1 = win1;
            game.state2 = win2;
```

**Review by SCDerox — CHANGES_REQUESTED** [review]
Thanks so much for your contribution - love the module.
I found some small (mostly style) issues. Please find them in the review comments. After fixing them, please request a re-review and we'll be ready to merge your PR.

Additionally, certain patterns in rock-paper-scissor.js seem to repeat multiple times (such as win calculation for users/bots). This is not something we require contributors to fix, so this is only a suggestion on how you could improve maintainability of your code in the future.

**Review by SCDerox — APPROVED** [review]
Thank you for your contribution, your module will be available on SCNX shortly.

**Pull request merged**